crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -200,13 +200,23 @@ def sanity_checks
   end
 
   def upgrade
-    respond_to do |format|
-      format.html do
-        # redirect to the upgrade ui wizard
-        redirect_to "/upgrade"
-      end
-      format.json do
-        render json: { error: I18n.t("error.during_upgrade") }, status: :service_unavailable
+    if request.post?
+      respond_to do |format|
+        format.html do
+          render "errors/during_upgrade", status: :service_unavailable
+        end
+        format.json do
+          render json: { error: I18n.t("error.during_upgrade") }, status: :service_unavailable
+        end
+      end
+    else
+      respond_to do |format|
+        format.html do
```

Comment:
AFAICS this is needless code duplication - you could just keep the original structure and put the `request.post?` conditional inside the `format.json` block.  Am I missing something?